backend/app/store/service/service.go

```diff
@@ -890,6 +902,44 @@ func (s *DataStore) Last(siteID string, limit int, since time.Time, user store.U
 	return s.alterComments(comments, user), nil
 }
 
+// Search executes search query and enriches the result with comment content
+func (s *DataStore) Search(siteID, query, sortBy string, limit int, skip int) ([]store.Comment, uint64, error) {
+	if s.SearchService == nil {
+		return []store.Comment{}, 0, fmt.Errorf("search service is not enabled")
+	}
+	req := &search.Request{
+		SiteID: siteID,
+		Query:  query,
+		Limit:  limit,
+		Offset: skip,
+		SortBy: sortBy,
+	}
+
+	seqrchRes, err := s.SearchService.Search(req)
+	if err != nil {
+		return nil, 0, fmt.Errorf("search failed: %w", err)
+	}
+
+	comments := make([]store.Comment, len(seqrchRes.Keys))
+	for i, docKey := range seqrchRes.Keys {
+		// ignore error because comment can be deleted from the store
+		comments[i], _ = s.Get(docKey.Locator, docKey.ID, nonAdminUser)
```

Comment:
this one is unclear. if Get doesn't return a valid comment, why we even put it to result? Maybe should be conditional append ?

Reply:
We do not delete anything from the search index but expect to have an error for the deleted comments. But this code was a bit wrong, we need to skip it, and now it's done.